### PR TITLE
libdogecoin: added moon

### PR DESCRIPTION
**Pull request by qlpqlp**
much moon phases  🌚🌓🌗🌝 :P

### CODE REVIEW

File: src/moon.c [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    case 4:
+      return "🌝";               // Full Moon DownTrend / No Discounts
+    case 5:
+      return "🐶";               // Show Doge wen no moon
+    case 6:
+      return "🌗";               // Third Quarter
+    case 7:
+      return "🐶";               // Show Doge wen no moon
+  }
+}
```
Comment (xanimo) [inline src/moon.c:78]: New line

**Review by xanimo — COMMENTED** [review]
This looks super cool, would you be willing to add to `Makefile.am`, `CMakeLists.txt` and perhaps write a test to assert your results?
```
touch test/moon_tests.c
```
and add to `test/unittester.c`?

No worries if not. Thank you regardless!

File: src/moon.c (hunk 1 above)
Comment (qlpqlp) [reply src/moon.c:78]: > This looks super cool, would you be willing to add to `Makefile.am`, `CMakeLists.txt` and perhaps write a test to assert your results?
> 
> ```
> touch test/moon_tests.c
> ```
> 
> and add to `test/unittester.c`?
> 
Never done it, but will try to do eheh :P

Comment (xanimo) [reply src/moon.c:78]: Let me know if you have questions

Comment (qlpqlp) [reply src/moon.c]: > Let me know if you have questions

lol, need some guidance, sorry :P

Comment (xanimo) [reply src/moon.c]: No prob. I'm going to check it out shortly, this looks p good though!

Comment (xanimo) [reply src/moon.c]: Alright so I struggled to come up with something re your tests or what exactly we should be testing there so as long as it's (`char* moon`) not returning false or 0 I think we should be fine? If it's based on current timing I don't think testing the previous way is the right path to go so I'd like to work with you on what we could come up with.

https://github.com/xanimo/libdogecoin/commits/e980e02bb5499c81413ef7ccc83adc98be5cfa24

Outside of that, I'd like to get the rest of the commits cleaned up so do you know how to interactive rebase?

Comment (qlpqlp) [reply src/moon.c]: Dono why is failing compiling lol 
Wen u have time to see why I appreciate, I'm a litle stuck :)
image

Comment (xanimo) [reply src/moon.c]: Hey, so if you can run the following and force push should be fine:
```
git checkout 0.1.2-dev
git branch -m 0.1.2-dev-moon-backup
git fetch upstream
git checkout -b 0.1.2-dev upstream/0.1.2-dev
```

Then just copy the files from: https://github.com/xanimo/libdogecoin/commit/8d262ac27327994a552220ce299c07f7f041ad16

And then:
```
git push -f origin 0.1.2-dev
```
And we should be good <3

Comment (qlpqlp) [reply src/moon.c:79]: So ardently the problem was on my test lol

Much thanks for taking your time to look at everything, teach me an even code for this to be in Libdogecoin :)

Comment (xanimo) [reply src/moon.c]: Alright my friend, apologies we have to do this one more time:
```
git checkout 0.1.2-dev
git fetch upstream
git reset --hard HEAD~1
```
Add the following changes (copy/paste):
https://github.com/xanimo/libdogecoin/commit/7ac420b57c2ea0160fcf89aaeca66f149226d645

Then:
```
git add .
git commit
git push -f origin 0.1.2-dev
```

**Comment by qlpqlp** [comment]
Added a new PR: https://github.com/dogecoinfoundation/libdogecoin/pull/98

**Pull request closed without merging**